### PR TITLE
feat: Display system message when message was partially sent [FS-1570]

### PR DESCRIPTION
**Pull request by atomrc**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-1570" title="FS-1570" target="_blank">Sub-taskFS-1570</a>  [web] Show system message indicating who will receive the message later
  </td></table>
  <br />
 

First batch of handling sending errors when one federated backend is offline. 

image
image
image

### CODE REVIEW

File: src/script/components/MessagesList/Message/ContentMessage/ContentMessage.tsx [hunk 1]
```diff
@@ -220,6 +230,13 @@ const ContentMessageComponent: React.FC<ContentMessageProps> = ({
           />
         ))}
 
+        {failedToSend && (
+          <FailedToSendWarning
+            failedToSend={failedToSend}
+            knownUsers={conversation.allUserEntities.map(user => ({id: user.qualifiedId, name: user.username()}))}
```
Comment (przemvs) [inline src/script/components/MessagesList/Message/ContentMessage/ContentMessage.tsx]: Maybe store it in variable and pass variable as props, WDYT?

File: src/script/components/MessagesList/Message/ContentMessage/FailedToSendWarning/FailedToSendWarning.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                  </span>
+                ))
+                .reduce((prev, element) => {
+                  return prev.length === 0 ? [element] : [...prev, ', ', element];
+                }, [] as any[])}
+              {` ${t('messageFailedToSendWillReceive')}`}
+            </p>
+          )}
+          <Button type="button" variant={ButtonVariant.TERTIARY} onClick={() => setIsOpen(!isOpen)}>
+            {isOpen ? 'Hide details' : 'Show details'}
```
Comment (przemvs) [inline src/script/components/MessagesList/Message/ContentMessage/FailedToSendWarning/FailedToSendWarning.tsx]: Translation missing ?

Comment (atomrc) [reply src/script/components/MessagesList/Message/ContentMessage/FailedToSendWarning/FailedToSendWarning.tsx]: good catch

File: src/script/components/MessagesList/Message/ContentMessage/ContentMessage.tsx (hunk 1 above)
Comment (atomrc) [reply src/script/components/MessagesList/Message/ContentMessage/ContentMessage.tsx]: sure

File: src/script/components/MessagesList/Message/ContentMessage/FailedToSendWarning/FailedToSendWarning.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                    {user.name}
+                  </span>
+                ))
+                .reduce((prev, element) => {
+                  return prev.length === 0 ? [element] : [...prev, ', ', element];
+                }, [] as any[])}
+              {` ${t('messageFailedToSendWillReceive')}`}
+            </p>
+          )}
+          <Button type="button" variant={ButtonVariant.TERTIARY} onClick={() => setIsOpen(!isOpen)}>
```
Comment (PatrykBuniX) [inline src/script/components/MessagesList/Message/ContentMessage/FailedToSendWarning/FailedToSendWarning.tsx]: It's not very safe to toggle the state this way. When we want to set a new state based on the previous one we should use callback instead: `setIsOpen(prev => !prev)`

File: src/script/components/MessagesList/Message/ContentMessage/FailedToSendWarning/FailedToSendWarning.tsx (hunk 2 above)
Comment (PatrykBuniX) [inline src/script/components/MessagesList/Message/ContentMessage/FailedToSendWarning/FailedToSendWarning.tsx]: Hardcoded content spotted 👁️

File: src/script/components/MessagesList/Message/ContentMessage/FailedToSendWarning/FailedToSendWarning.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+            <p css={warning}>
+              {namedUsers
+                .map(user => (
+                  <span data-uie-name="recipient" data-uie-value={user.id.id} key={user.id.id}>
+                    {user.name}
+                  </span>
+                ))
+                .reduce((prev, element) => {
+                  return prev.length === 0 ? [element] : [...prev, ', ', element];
+                }, [] as any[])}
```
Comment (PatrykBuniX) [inline src/script/components/MessagesList/Message/ContentMessage/FailedToSendWarning/FailedToSendWarning.tsx]: Can we get rid of this `any`?

Comment (atomrc) [reply src/script/components/MessagesList/Message/ContentMessage/FailedToSendWarning/FailedToSendWarning.tsx]: sure thing, good catch

File: src/script/components/MessagesList/Message/ContentMessage/FailedToSendWarning/FailedToSendWarning.tsx (hunk 3 above)
Comment (atomrc) [reply src/script/components/MessagesList/Message/ContentMessage/FailedToSendWarning/FailedToSendWarning.tsx]: True that, nice one

**Review by PatrykBuniX — APPROVED** [review]
Nice 👍

File: src/script/components/MessagesList/Message/ContentMessage/ContentMessage.tsx (hunk 1 above)
Comment (atomrc) [reply src/script/components/MessagesList/Message/ContentMessage/ContentMessage.tsx]: Actually, maybe we won't even need this props at all. It's in discussion with product currently

**Review by przemvs — APPROVED** [review]

File: src/script/components/MessagesList/Message/ContentMessage/FailedToSendWarning/FailedToSendWarning.styles.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {CSSObject} from '@emotion/react';
+
+export const warning: CSSObject = {color: 'var(--danger-color)'};
```
Comment (V-Gira) [inline src/script/components/MessagesList/Message/ContentMessage/FailedToSendWarning/FailedToSendWarning.styles.ts]:
```suggestion
export const warning: CSSObject = {color: 'var(--danger-color)', fontSize: 'var(--font-size-small)'};
```

this is going to be shared by all warnings, we could maybe change the file structure to `.../Message/ContentMessage/Warnings/FailedToSendWarning/FailedtToSendWarning.tsx` with this specific style at `.../ContentMessage/Warnings/Warnings.styles.ts` ?

Comment (atomrc) [reply src/script/components/MessagesList/Message/ContentMessage/FailedToSendWarning/FailedToSendWarning.styles.ts]: That's good call :)

**Review by V-Gira — APPROVED** [review]

**Pull request merged**